### PR TITLE
update host.md

### PR DESCRIPTION
**Pull request by uhhmed**

### CODE REVIEW

File: docs/host.md [hunk 1]
```diff
@@ -28,19 +28,39 @@ User Name: testuser
 
 ## Read Python Environment
 
-TBD
+1. Click the second icon from the left (python icon) at the bottom right corner of the "Host Details" page.
```
Comment (ZihengSun) [inline docs/host.md:31]:
```suggestion
1. Click the python icon button in the toolbar of the shown "Host Details" page.
```

File: docs/host.md [hunk 2]
```diff
@@ -28,19 +28,39 @@ User Name: testuser
 
 ## Read Python Environment
```
Comment (ZihengSun) [inline docs/host.md:30]: How do people see the "Host Details" page?

File: docs/host.md [hunk 3]
```diff
@@ -28,19 +28,39 @@ User Name: testuser
...
 
-TBD
+1. Click the second icon from the left (python icon) at the bottom right corner of the "Host Details" page.
+
+2. Input the `Host User Password` of your account on the server in the shown dialog pop up.
```
Comment (ZihengSun) [inline docs/host.md:33]: What about the Localhost? Where do they find password for localhost?

File: docs/host.md [hunk 4]
```diff
@@ -28,19 +28,39 @@ User Name: testuser
...
+
+3. An "Environment List" section will appear below the "Host Details" section showing all the available python environments on the connected server.
```
Comment (ZihengSun) [inline docs/host.md:35]: Are they really `all` available environments? Does it read both `pipenv` and `conda`? what about `virtualenv`?

File: docs/host.md [hunk 5]
```diff
@@ -28,19 +28,39 @@ User Name: testuser
...
 
 ## File Browser
 
-TBD
+1. Click the second icon from the right (next to the red minus button) in the "Host Details" page.
```
Comment (ZihengSun) [inline docs/host.md:39]: Same problem. How do people access the `Host Details`.

Comment (ZihengSun) [inline docs/host.md:39]:
```suggestion
1. Click the sitemap icon (the one with branches) button in the "Host Details" toolbar.
```

File: docs/host.md [hunk 6]
```diff
@@ -28,19 +28,39 @@ User Name: testuser
...
+
+2. Input the `Host User Password` of your account on the server in the shown dialog pop up.
```
Comment (ZihengSun) [inline docs/host.md:41]: Same problem as above.

File: docs/host.md [hunk 7]
```diff
@@ -28,19 +28,39 @@ User Name: testuser
...
+
+3. Click Confirm.
```
Comment (ZihengSun) [inline docs/host.md:43]: What happens after `confirm`? How do people know they do it right?

File: docs/host.md [hunk 8]
```diff
@@ -28,19 +28,39 @@ User Name: testuser
...
 
 ## File Uploader
 
-TBD
+1. Click the upload file icon (the middle icon) at the bottom right corner of the "Host Details" page.
```
Comment (ZihengSun) [inline docs/host.md:47]:
```suggestion
1. Click the uploader icon button in the "Host Details" toolbar.
```

File: docs/host.md [hunk 9]
```diff
@@ -28,19 +28,39 @@ User Name: testuser
...
+
+2. Input the `Host Password` of your account on the server in the shown dialog pop up.
```
Comment (ZihengSun) [inline docs/host.md:49]: same

File: docs/host.md [hunk 10]
```diff
@@ -28,19 +28,39 @@ User Name: testuser
...
+
+3. A file upload section will appear below the "Host Details" section with drag and drop or file browser capability.
```
Comment (ZihengSun) [inline docs/host.md:51]: Better have a screenshot.

File: docs/host.md [hunk 11]
```diff
@@ -28,19 +28,39 @@ User Name: testuser
...
+
+4. Choose the desired file to upload, and click the start button below the "File Uploader Section" to initiate the upload.
```
Comment (ZihengSun) [inline docs/host.md:53]:
```suggestion
4. Click on the `Open the File Browser` button to choose files, and click `Start` to initiate the uploading.
```

File: docs/host.md [hunk 12]
```diff
@@ -28,19 +28,39 @@ User Name: testuser
...
+
+5. A progress bar with a status above will indicate the progress of the upload.
```
Comment (ZihengSun) [inline docs/host.md:55]: What is next? What happens when uploading finishes? Where did the file go? 

A tutorial should have beginning and end. Don't leave without giving people an end.

File: docs/host.md [hunk 13]
```diff
@@ -28,19 +28,39 @@ User Name: testuser
...
 
 ## Edit a Host
 
-TBD
+1. Click the edit icon (first icon) at the bottom right corner of the "Host Details" page.
+
+2. Change the desired fields.
```
Comment (ZihengSun) [inline docs/host.md:61]:
```suggestion
2. Change the field values as you wish. 
Note: The Host Id is not editable. 
```

File: docs/host.md [hunk 14]
```diff
@@ -28,19 +28,39 @@ User Name: testuser
...
+
+3. Click the same edit icon as the first step. The changes have now been updated.
```
Comment (ZihengSun) [inline docs/host.md:63]: No need to click the edit button every time. Once you finish, click any place outside the text fields will trigger the save action. It is listening to the focus event.

**Review by ZihengSun — CHANGES_REQUESTED** [review]

File: docs/host.md (hunk 2 above)
Comment (uhhmed) [reply docs/host.md:30]: An additional step is added to mention how to get here.

File: docs/host.md (hunk 3 above)
Comment (uhhmed) [reply docs/host.md]: A note has been added at every occurrence of password input in all sections with a link to the password instructions in readme.md

File: docs/host.md (hunk 4 above)
Comment (uhhmed) [reply docs/host.md]: A "PyEnv" explicit callout has been added.

File: docs/host.md (hunk 5 above)
Comment (uhhmed) [reply docs/host.md]: An additional step is added to mention how to get here.

File: docs/host.md (hunk 6 above)
Comment (uhhmed) [reply docs/host.md:47]: An additional step is added to mention how to get here.

File: docs/host.md (hunk 7 above)
Comment (uhhmed) [reply docs/host.md]: Expanded the last step to mention the appropriate section appearing when the function executes correctly

File: docs/host.md (hunk 9 above)
Comment (uhhmed) [reply docs/host.md:59]: A note has been added at every occurrence of password input in all sections with a link to the password instructions in readme.md

File: docs/host.md (hunk 10 above)
Comment (uhhmed) [reply docs/host.md:63]: A screenshot has been added and uploaded.

File: docs/host.md (hunk 12 above)
Comment (uhhmed) [reply docs/host.md:68]: Additional step is added to mention what happens when the upload is complete and where the file is.

File: docs/host.md (hunk 14 above)
Comment (uhhmed) [reply docs/host.md:80]: This doesn't work with me. if i click outside any of the fields, or even the right panel, nothing happens. only when the edit icon is clicked is when the values are actually updated.

File: docs/host.md [hunk 15]
```diff
@@ -28,19 +28,56 @@ User Name: testuser
...
+
+5. A progress bar with a status above will indicate the progress of the upload.
+
+6. The status with the message "Upload Complete" in green will show when the file is successfully uploaded, and the file is uploaded to the home directory of the logged in user.
+
 
 ## Edit a Host
 
-TBD
+1. Click the edit icon (first icon) at the bottom right corner of the "Host Details" page.
```
Comment (ZihengSun) [inline docs/host.md:75]:
```suggestion
1. Click the edit icon (first icon) in the `Host Details` page.
```

File: docs/host.md [hunk 16]
```diff
@@ -28,19 +28,56 @@ User Name: testuser
...
+
+2. Change the field values as you wish. 
+Note: The Host Id is not editable. 
+
+3. Click the same edit icon as the first step. The changes have now been updated.
```
Comment (ZihengSun) [inline docs/host.md:80]:
```suggestion
3. Click the edit icon button again. The changes should be updated.
```

File: docs/host.md (hunk 14 above)
Comment (ZihengSun) [reply docs/host.md:80]: That is how the fields in process/workflow pages work. It is fine. Just describe how it works in Host page.

File: docs/host.md [hunk 17]
```diff
@@ -28,19 +28,56 @@ User Name: testuser
...
-TBD
+1. Select the desired host under `Hosts` on the right panel.
+
+2. Click the python icon button in the toolbar of the shown "Host Details" page.
+
+2. Input the `Host User Password` of your account on the remote server in the shown dialog pop up.
+
+Note: if the selected server is localhost, password instructions are discussed [here](https://github.com/ESIPFed/Geoweaver#reset-password-for-localhost "Reset Password for Localhost")
+
+3. An "Environment List" section will appear below the "Host Details" section showing all the available PyEnv python environments on the connected server.
```
Comment (ZihengSun) [inline docs/host.md:39]:
```suggestion
3. An `Environment List` section will appear below the `Host Details` section showing all the available PyEnv python environments on the connected server.
```

**Review by ZihengSun — CHANGES_REQUESTED** [review]
Change all " to `

**Pull request merged**